examples/fluids/problems/stg_shur14.c

```diff
@@ -255,7 +260,9 @@ PetscErrorCode SetupStg(const MPI_Comm comm, const DM dm, ProblemData problem, U
 
     PetscInt nmax = 3, faces[3];
     PetscCall(PetscOptionsGetIntArray(NULL, NULL, "-dm_plex_box_faces", faces, &nmax, NULL));
-    global_stg_ctx->dx = given_stg_dx ? stg_dx : domain_size[0] / faces[0];
+    global_stg_ctx->dx      = given_stg_dx ? stg_dx : domain_size[0] / faces[0];
+    global_stg_ctx->dyScale = given_stg_dyScale ? stg_dyScale : 1;
+    global_stg_ctx->dz      = given_stg_dz ? stg_dz : domain_size[2] / faces[2];
```

Comment:
@KennethEJansen Any qualms with the default being to setting the dz spacing based on the element size rather than `domain_size[2] / faces[2]`, which assumes a PETSc box mesh?

Reply:
The issue is that the size SHOULD NOT be determined solely by the mesh for a higher order method.  I switched to this form  so that I had the ability to scale the element "measure" for polynomial order.

We had discussed a default scaling of 1/p

Reply:
Right, but using the domain_size and faces is worse than just going based on the element size (while also being the same).

Can we:
- Remove `-stg_dy_scale`
- Add `-stg_element_scale` to affect all the directions equally
- Have `-stg_dx` and `-stg_dz` represent actual element size (rather than nodal distance)

> the element "measure"

I was thinking about what we want this quantity to be and it's really just the inverse of the Nyquist frequency for the element, which is exactly equal to $h/p$. So a $1/p$ scale factor default really does make the most sense I think.

Reply:
I don't know that I believe STG  should always be h/p.  While I am the vertex spacing can't be right for all p. I am further not sure that it should the same for all directions but certainly that is an improvement and possibly the best that can be done so......

I agree to the first 2 of your bullets but don't follow the third.  What is the actual element size to you?
I don't follow the fourth bullet either

The point after "measure" quote seems to argue against letting the user  scale, forcing it to be 1/p which I don't agree with.

Can we say,
Currently the code can and does compute element lengths in the wall normal and span direction from the in-plane metric tensor. 
We will refer to these as stg_d_wn and stg_d_sw  These are element sizes (akin to corner vertex separations or bounding box dimensions. 
At some later date we expect to be able to compute the plane-normal element size stg_d_pn.
Until that time, the user MUST provide stg_d_pn as a physical length of elements near the inflow.
User is free to provide a scaling  for those THREE element lengths.  If no scaling is provided 1/p will be a default.


I am ok leaving _wn=y,  _sw=z, and _pn=x for now  if it touches too much code to change quickly, but we really should not have assumed a coordinate system for the inflow.

Reply:
Your suggestion is what I was trying to explain. I can get on that.

> I am ok leaving _wn=y, _sw=z, and _pn=x for now if it touches too much code to change quickly, but we really should not have assumed a coordinate system for the inflow.

We have those kind of bad-but-ok coordinate system assumptions all over the place right now (IDL, spanwise statistics). A wall-oriented coordinate system is better and would be pretty easily implemented, but we need distance-to-wall to do that generically. I'm not sure if you had something else in mind.

Reply:
Updated now.

Reply:
If we consider the metric tensor on the STG inflow plane, its eigenvalues are a coordinate-system-independent length scale for the element. Should we use that by default?

Reply:
That is the current default for the wall-parallel directions, but we can't get that for the boundary-normal direction, which requires wall-normal gradients at the boundary for the metric tensor. I'm currently working on copying that over from Ratel, so hopefully that'll be done soon :tm:. But for right now the element length in that direction needs to be given explicitly.

Reply:
I think you mean "requires (STG) boundary-normal gradients", but some time ago, we discussed using the spanwise (longest length scale in STG boundary plane) and multiplying by a factor (2.5 could be default, which matches the 6-15 or 12-30) to get the STG boundary-normal length scale.

Reply:
Yeah, I meant boundary normal.

I had a PR for the dx scaling factor in #1415, but Ken went with the more direct way for his branch. Regardless of which method (directly specifying $\Delta x$ or setting it as a scale of $\Delta z$), they should be superseded once the boundary-normal gradient in merged in.